src/digest/sha1.rs

```diff
@@ -33,16 +33,18 @@ fn parity(x: W32, y: W32, z: W32) -> W32 {
 type State = [W32; CHAINING_WORDS];
 const ROUNDS: usize = 80;
 
-pub(super) extern "C" fn block_data_order(
-    state: &mut super::State,
-    data: *const u8,
-    num: c::size_t,
-) {
-    let state = unsafe { &mut state.as32 };
-    let state: &mut State = (&mut state[..CHAINING_WORDS]).try_into().unwrap();
-    let data = data as *const [<W32 as Word>::InputBytes; 16];
-    let blocks = unsafe { core::slice::from_raw_parts(data, num) };
-    *state = block_data_order_(*state, blocks)
+versioned_extern_def! {
```

Comment:
Does this really need to be versioned? Itn's not a `no_mangle` function.

Reply:
hmm I'm not sure if mangling is the issue here. if we link together two different versions of ring, does mangling guarantee that they won't conflict? my (limited) understanding of Rust symbol mangling is that it's to avoid conflicts across modules, not across versions. I'll do some testing to see what happens if these don't get prefixed.